tracing-subscriber/src/fmt/format.rs

```diff
@@ -280,8 +309,61 @@ where
     }
 }
 
-// === impl FormatFields ===
+#[cfg(feature = "json")]
+impl<N, T> FormatEvent<N> for Format<Json, T>
+where
+    N: for<'writer> FormatFields<'writer>,
+    T: FormatTime,
+{
+    fn format_event(
+        &self,
+        ctx: &span::Context<'_, N>,
+        writer: &mut dyn fmt::Write,
+        event: &Event<'_>,
+    ) -> fmt::Result {
+        let mut timestamp = String::new();
+        self.timer.format_time(&mut timestamp)?;
+
+        #[cfg(feature = "tracing-log")]
+        let normalized_meta = event.normalized_metadata();
+        #[cfg(feature = "tracing-log")]
+        let meta = normalized_meta.as_ref().unwrap_or_else(|| event.metadata());
+        #[cfg(not(feature = "tracing-log"))]
+        let meta = event.metadata();
+
+        let mut visit = || {
+            let mut serializer = Serializer::new(WriteAdaptor::new(writer));
+            let mut serializer = serializer.serialize_map(None)?;
+
+            serializer.serialize_key("timestamp")?;
+            serializer.serialize_value(timestamp.trim())?;
+            serializer.serialize_key("level")?;
+            serializer.serialize_value(&format!("{}", meta.level()))?;
```

Comment:
Could these be serialized instead using https://tracing.rs/tracing_serde/#structs?

Reply:
No worries!

Reply:
No worries!